src/DotLiquid.Tests/HashTests.cs

```diff
@@ -71,26 +80,38 @@ public void TestMapperCacheShouldCacheSeperateNamespaces()
 
         private void IncludeBaseClassPropertiesOrNot(bool includeBaseClassProperties)
         {
+            var TestClassOverridablePropValue = "TestClassOverridablePropValue";
             var TestClassPropValue = "TestClassPropValueValue";
+            var TestMiddleClassPropValue = "TestMiddleClassPropValue";
             var TestBaseClassPropValue = "TestBaseClassPropValue";
 
-            var value = Hash.FromAnonymousObject(new TestClass()
+            var value = Hash.FromAnonymousObject(new TestChildClass()
             {
                 TestClassProp = TestClassPropValue,
-                TestBaseClassProp = TestBaseClassPropValue
+                TestMiddleClassProp = TestMiddleClassPropValue,
+                TestBaseClassProp = TestBaseClassPropValue,
+                TestOverridableProp = TestClassOverridablePropValue
             }, includeBaseClassProperties);
 
             Assert.AreEqual(
                 TestClassPropValue,
-                value[nameof(TestClass.TestClassProp)]);
+                value[nameof(TestChildClass.TestClassProp)]);
+
+            Assert.AreEqual(
+                includeBaseClassProperties ? TestMiddleClassPropValue : null,
+                value[nameof(TestMiddleClass.TestMiddleClassProp)]);
 
             Assert.AreEqual(
                 includeBaseClassProperties ? TestBaseClassPropValue : null,
-                value[nameof(TestClass.TestBaseClassProp)]);
+                value[nameof(TestChildClass.TestBaseClassProp)]);
+
+            Assert.AreEqual(
```

Comment:
might want to keep this together with the other "ChildClass" property test. Would also help to comment
// Properties defined in TestChildClass should be exposed regardless of includeBaseClassProperties

and similarly for middle/base class tests